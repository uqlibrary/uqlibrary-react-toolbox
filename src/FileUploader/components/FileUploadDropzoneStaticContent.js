import React from 'react';
import PropTypes from 'prop-types';
import FontIcon from 'material-ui/FontIcon';

const FileUploadDropzoneStaticContent = ({locale}) => (
    <div className="columns file-instructions">
        <div className="column">
            {locale.fileUploadRestrictionHeading}
            {locale.fileUploadRestrictions}
        </div>
        <div className="column upload-instructions">
            <FontIcon className="material-icons">cloud_upload</FontIcon>
            {locale.fileUploadInstruction}
        </div>
    </div>
);

FileUploadDropzoneStaticContent.propTypes = {
    locale: PropTypes.object
};

export default FileUploadDropzoneStaticContent;
