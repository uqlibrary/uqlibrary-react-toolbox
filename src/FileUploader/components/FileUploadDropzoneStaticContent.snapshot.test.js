jest.dontMock('./FileUploadDropzoneStaticContent');

import React from 'react';
import FileUploadDropzoneStaticContent from './FileUploadDropzoneStaticContent';

function setup(testProps, isShallow = true) {
    const props = {
        ...testProps
    };

    return getElement(<FileUploadDropzoneStaticContent {...props} />, isShallow);
}

describe('Component FileUploadDropzoneStaticContent', () => {
    it('should render with default setup', () => {
        const props = {
            txt: {
                fileUploadRestrictionHeading: 'Restrictions for file upload',
                fileUploadRestrictions: 'No restrictions applies to Tests!!!',
                fileUploadInstruction: 'Upload whatever you like :)'
            }
        };
        const wrapper = setup({...props}, false);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
