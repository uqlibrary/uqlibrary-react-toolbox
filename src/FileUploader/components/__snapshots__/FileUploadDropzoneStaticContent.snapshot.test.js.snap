// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FileUploadDropzoneStaticContent should render with default setup 1`] = `
<FileUploadDropzoneStaticContent
  txt={
    Object {
      "fileUploadInstruction": "Upload whatever you like :)",
      "fileUploadRestrictionHeading": "Restrictions for file upload",
      "fileUploadRestrictions": "No restrictions applies to Tests!!!",
    }
  }
>
  <div
    className="columns file-instructions"
  >
    <div
      className="column"
    >
      Restrictions for file upload
      No restrictions applies to Tests!!!
    </div>
    <div
      className="column upload-instructions"
    >
      <FontIcon
        className="material-icons"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="material-icons"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fontSize": 24,
              "muiPrepared": true,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
            }
          }
        >
          cloud_upload
        </span>
      </FontIcon>
      Upload whatever you like :)
    </div>
  </div>
</FileUploadDropzoneStaticContent>
`;
